packages/xy-chart/src/utils/computeCirclePack.js

```diff
@@ -41,7 +43,11 @@ function packCircles(data, xScale, getSize = d => d.size || 4) {
   function xpackBestPlace(startn, node) {
     const goodnodes = [];
     for (let p = startn.nextPack; p !== startn; p = p.nextPack) {
-      if (!(p.px + p.r < node.x - node.r || p.px - p.r > node.x + node.r)) {
+      if (!(p.px + p.r < node.x - node.r ||
+          p.px - p.r > node.x + node.r ||
+          p.px - p.r - node.r < globalMinX ||
```

Comment:
do you think that some of the circles still fall outside the bounds because this is a series of `||` conditions instead of `&&`?